### PR TITLE
RELATED: RAIL-3428 - First working version of the add command

### PR DESCRIPTION
**Pull request by lupko**
See commits

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5736/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7472/ : SUCCESS in 15m 25s

**Review by plavemar — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5738/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7476/ : FAILURE in 6m 42s

File: tools/dashboard-plugin-template/webpack.config.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ module.exports = (_env, argv) => {
                     PORT: JSON.stringify(PORT),
                 }),
                 new Dotenv({
-                    silent: true, // we are ok with the .env file not being there, do not warn about it (so that CI works)
+                    silent: false, // we are ok with the .env file not being there, do not warn about it (so that CI works)
```
Comment (no23reason) [inline tools/dashboard-plugin-template/webpack.config.js:146]: The comment is no longer valid. But I suspect this is what is nuking your build currently.

File: libs/sdk-backend-bear/src/backend/index.ts [hunk 2]
```diff
@@ -343,7 +343,10 @@ export class BearBackend implements IAnalyticalBackend {
 
     private getAuthenticationContext = (): IAuthenticationContext => ({ client: this.sdk, backend: this });
 
-    private triggerAuthentication = (reset = false): Promise<IAuthenticatedPrincipal> => {
+    private triggerAuthentication = (
+        reset = false,
+        errorConverter: ErrorConverter = convertApiError,
```
Comment (no23reason) [inline libs/sdk-backend-bear/src/backend/index.ts:348]: What is the use for this parameter? I don't see it used anywhere (passed a value).

File: libs/sdk-backend-tiger/src/backend/index.ts [hunk 3]
```diff
@@ -300,7 +300,10 @@ export class TigerBackend implements IAnalyticalBackend {
         };
     };
 
-    private triggerAuthentication = (reset = false): Promise<IAuthenticatedPrincipal> => {
+    private triggerAuthentication = (
+        reset = false,
+        errorConverter: ErrorConverter = convertApiError,
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/index.ts:305]: What is the use for this parameter? I don't see it used anywhere (passed a value).

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: tools/dashboard-plugin-template/webpack.config.js (hunk 1 above)
Comment (lupko) [reply tools/dashboard-plugin-template/webpack.config.js]: interesting, i don't even remember turning it to false :) anyway, i've made it silent & have two entries one for default .env and one for .env.secrets

File: libs/sdk-backend-bear/src/backend/index.ts (hunk 2 above)
Comment (lupko) [reply libs/sdk-backend-bear/src/backend/index.ts:348]: actually, looking through this i realized i made the fix unnecessary dangerous. will prepare rework shortly

File: libs/sdk-backend-tiger/src/backend/index.ts (hunk 3 above)
Comment (lupko) [reply libs/sdk-backend-tiger/src/backend/index.ts:305]: actually, looking through this i realized i made the fix unnecessary dangerous. will prepare rework shortly

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

File: libs/sdk-backend-bear/src/backend/index.ts (hunk 2 above)
Comment (lupko) [reply libs/sdk-backend-bear/src/backend/index.ts]: fix reworked

File: libs/sdk-backend-tiger/src/backend/index.ts (hunk 3 above)
Comment (lupko) [reply libs/sdk-backend-tiger/src/backend/index.ts]: fix reworked

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1888

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1632/ : SUCCESS in 3m 17s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1912/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/709/ : SUCCESS in 14m 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7481/ : SUCCESS in 21m 26s

**Pull request merged**